honeycomb-core/src/cmap2/structure.rs

```diff
@@ -211,8 +211,9 @@ impl<T: CoordsFloat> CMap2<T> {
     #[must_use = "constructed object is not used, consider removing this function call"]
     pub(crate) fn new_with_attributes(
         n_darts: usize,
-        attr_storage_manager: AttrStorageManager,
+        mut attr_storage_manager: AttrStorageManager,
     ) -> Self {
+        attr_storage_manager.extend_storages(n_darts + 1);
```

Comment:
Not very clear: what `extend_storages` do and why `n_darts + 1`?

Reply:
I added comments & a little bit of doc on what's expected of the `CMap2::new_with_attributes` constructor in d67a185. Essentially, all storages are of null length before the call to `extend_storages`; The method is used to prepare all storages before they are accessed when manipulating the map. 

This is necessary because all attribute accesses in the `0..n_darts+1` range should be valid according to how we implement embedding (although accessed values may be `None`)